utility/user.middleware.js

```diff
@@ -0,0 +1,31 @@
+const users = require('../models/user')
+
+module.exports.byEmailAndMobile = async (req, res, next) => {
+    try{
+        const { email, mobile } = req.body;
+        const result = await users.findOne({$or:[{email:email},{mobile:mobile}]});
+        if(result) {
+            req.userResult = result;
+            return next();
+        }
+        return next();
+    } catch(err) {
+        console.log(err);
+    } 
+}
```

Comment:
Function name should be like emailAndMobileCheck and also return error from this function if email or mobile already exist and catch block should also return error not only just log error